middleware.js

```diff
@@ -0,0 +1,30 @@
+import { getToken } from "next-auth/jwt";
+import { NextResponse } from "next/server";
+
+export async function middleware(req) {
+  const session = await getToken({
+    req: req,
+    secret: process.env.NEXTAUTH_SECRET,
+  });
+  const reqPathName = req.nextUrl.pathname;
+
+  if (!session) {
+    if (reqPathName.startsWith("/api")) {
+      return NextResponse.json({}, { status: 401 });
+    }
+    return NextResponse.redirect(new URL("/auth/signin", req.url));
+  }
+
+  const username = session.username;
+  if (!process.env.ADMIN_USERS.includes(username)) {
+    if (reqPathName.startsWith("/api")) {
```

Comment:
Should this be `/api/admin`?

Reply:
ah we could add that but matcher config already does match for 
`  matcher: ["/admin/:path*", "/api/admin/:path*"]`
So i think it should be ok, but can update to /api/admin as well just to be sure